src/bridge/PublicitySyncer.ts

```diff
@@ -202,25 +202,25 @@ export class PublicitySyncer {
         const cli = this.ircBridge.getAppServiceBridge().getBot().getClient();
 
         // Update rooms to correct visibilities
-        const promises = roomIds.map((roomId) => {
-            const currentState = this.visibilityMap.roomVisibilities[roomId];
+        const promises = roomIds.map(async (roomId) => {
+            const currentState = await this.ircBridge.getStore().getRoomVisibility(roomId);
```

Comment:
Possible perf issue - what is the expected size of `roomIds` here? Could end up making loads of tiny `select`s when you could group them together then iterate.